### PR TITLE
IGNITE-16453 Try to optimize IgniteLongList

### PR DESCRIPTION
**Pull request by tkalkirill**
https://issues.apache.org/jira/browse/IGNITE-16453

### CODE REVIEW

File: modules/page-memory/src/main/java/org/apache/ignite/internal/pagememory/reuse/LongListReuseBag.java [hunk 1]
```diff
@@ -17,13 +17,13 @@
...
+import it.unimi.dsi.fastutil.longs.LongArrayList;
 import java.io.Externalizable;
-import org.apache.ignite.internal.util.IgniteLongList;
 
 /**
- * {@link IgniteLongList}-based reuse bag.
+ * {@link LongArrayList}-based reuse bag.
  */
-public final class LongListReuseBag extends IgniteLongList implements ReuseBag {
+public final class LongListReuseBag extends LongArrayList implements ReuseBag {
```
Comment (sashapolo) [inline modules/page-memory/src/main/java/org/apache/ignite/internal/pagememory/reuse/LongListReuseBag.java:26]: Why does this class have to extend `LongArrayList`? Doesn't it make sense to use composition here?

Comment (sashapolo) [reply modules/page-memory/src/main/java/org/apache/ignite/internal/pagememory/reuse/LongListReuseBag.java:26]: Also, would it make more sense to use `LongArrayFIFOQueue` here?

File: modules/page-memory/src/main/java/org/apache/ignite/internal/pagememory/tree/BplusTree.java [hunk 2]
```diff
@@ -4550,10 +4551,10 @@ public long pollFreePage() {
                 return 0L;
             }
 
-            if (freePages.getClass() == IgniteLongList.class) {
-                IgniteLongList list = ((IgniteLongList) freePages);
+            if (freePages.getClass() == LongArrayList.class) {
```
Comment (sashapolo) [inline modules/page-memory/src/main/java/org/apache/ignite/internal/pagememory/tree/BplusTree.java:4554]: why not `instanceOf`? Also, maybe a queue is more suitable here?

Comment (tkalkirill) [reply modules/page-memory/src/main/java/org/apache/ignite/internal/pagememory/tree/BplusTree.java:4554]: I think reference equality is used here for better performance, I need to think about the queue.

Comment (ibessonov) [reply modules/page-memory/src/main/java/org/apache/ignite/internal/pagememory/tree/BplusTree.java:4554]: I doubt that the performance difference is even measurable in this case

Comment (tkalkirill) [reply modules/page-memory/src/main/java/org/apache/ignite/internal/pagememory/tree/BplusTree.java:4554]: I fixed it, hope it's better.

File: modules/page-memory/src/main/java/org/apache/ignite/internal/pagememory/reuse/LongListReuseBag.java (hunk 1 above)
Comment (tkalkirill) [reply modules/page-memory/src/main/java/org/apache/ignite/internal/pagememory/reuse/LongListReuseBag.java]: Fixed it.

**Comment by tkalkirill** [comment]
@ibessonov @sashapolo Guys please look again.

**Review by sashapolo — APPROVED** [review]

**Review by ibessonov — APPROVED** [review]

**Pull request merged**